ipatests/test_integration/test_replica_promotion.py

```diff
@@ -1049,7 +1050,15 @@ def test_ipahealthcheck_hidden_replica(self):
                 srv,
                 failures_only=True
             )
-            assert returncode == 0
+            if (os_version[0] == 'fedora'
+                and pki_version < tasks.parse_version('11.1.0')) \
+                    or (os_version[0] == 'rhel' and os_version[1][0] == 8
+                        and pki_version < tasks.parse_version('10.12.0')) \
+                    or (os_version[0] == 'rhel' and os_version[1][0] == 9
+                        and pki_version < tasks.parse_version('11.0.4')):
+                pytest.xfail('https://pagure.io/freeipa/issue/8582')
+            else:
+                assert returncode == 0
```

Comment:
I'd rather use xfail_context as it actually checks that the test fails if pki version doesn't contain the fix (pytest.xfail _declares_ that the test fails, it doesn't check anything):
```
           pki_too_old =  (os_version[0] == 'fedora'
                and pki_version < tasks.parse_version('11.1.0')) \
                    or (os_version[0] == 'rhel' and os_version[1][0] == 8
                        and pki_version < tasks.parse_version('10.12.0')) \
                    or (os_version[0] == 'rhel' and os_version[1][0] == 9
                        and pki_version < tasks.parse_version('11.0.4'))
                
            with xfail_context(pki_too_old, 'https://pagure.io/freeipa/issue/8582'):
                assert returncode == 0
```